torchao/prototype/float8nocompile/test/fsdp_test.py

```diff
@@ -0,0 +1,97 @@
+######################################################################
+#
+# To run these unit tests, use the following command:
+#
+# torchrun --nproc_per_node=${NUM_GPUS} -m pytest test/fsdp_test.py
+#
+#######################################################################
+import os
+
+import pytest
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
```

Comment:
this is FSDP1.  Can we test FSDP2, which is the new, recently released version of FSDP with support for float8 all-gather?

https://github.com/pytorch/torchtitan/blob/main/docs/fsdp.md

IMO testing just FSDP2 is fine